src/vs/workbench/services/search/electron-browser/searchService.ts

```diff
@@ -84,6 +86,10 @@ export class DiskSearch implements ISearchResultProvider {
 		const client = new Client(FileAccess.asFileUri('bootstrap-fork', require).fsPath, opts);
 		const channel = getNextTickChannel(client.getChannel('search'));
 		this.raw = new SearchChannelClient(channel);
+
+		this.lifecycleService.onDidShutdown(() => {
```

Comment:
Don't we need similar code for the file watcher too?

Reply:
I did another sanity check:
Assuming that `src/vs/platform/files/node/diskFileSystemProvider.ts` lists all the available file watchers:
- `UnixWatcherService` extends `Disposable`, and registers the new `Client`, so it is automatically disposed.
- `WindowsWatcherService` implements `IDisposable`, and in the `dispose()` function, kills the child process.
- `NsfwWatcherService` does the same thing as `UnixWatcherService`
- `NodeJSWatcherService` extends `Disposable`, and registers the `watchFile()` and `watchFolder()` calls. It calls `doWatchNonRecursive()` in `watcher.ts`. I can look into that code if necessary, but for now, I am assuming that function creates and disposes the resources correctly.
- `DiskFileSystemProvider` extends `Disposable`, but I just noticed that it might be leaked in a few cases. I'll create a new PR for this, because it affects a few other files.

Reply:
Ah yeah good points 👍 . In fact, the disk file system provider is properly disposed from here:

https://github.com/microsoft/vscode/blob/8215a110a9469d99f42abe9dfcfa44e19282aa01/src/vs/workbench/electron-sandbox/shared.desktop.main.ts#L127-L127

I think the node.js watcher service - even if it were not disposed - is not an issue because that one does not use child processes.